src/ui/AttachResult/AttachResult.ts

```diff
@@ -0,0 +1,180 @@
+import {
+    Component,
+    ComponentOptions,
+    IComponentBindings,
+    $$,
+    IQueryResult,
+    Initialization,
+    IAnalyticsCaseAttachMeta,
+    IAnalyticsCaseDetachMeta,
+    analyticsActionCauseList
+} from "coveo-search-ui";
+import * as PaperclipIcon from "../../icons/paperclip.svg";
+
+export interface IAttachResultOptions {
+    /** Specifies the tooltip displayed when the result is not attached. */
+    attachCaption?: string;
+    /** Specifies the tooltip displayed when the result is already attached. */
+    detachCaption?: string;
+    /** The function that is called when the user wants to attach a result. */
+    attach: (queryResult: IQueryResult) => Promise<void>;
+    /** The function called when the user wants to un-link a result. */
+    detach: (queryResult: IQueryResult) => Promise<void>;
+    /** Optional function to check the initial state of the component. */
+    isAttached?: (queryResult: IQueryResult) => Promise<boolean>;
+}
```

Comment:
If you ever want to provide an option for tooltip positioning, instead of it always being at the bottom, for the Quickview component we use PopperJs
https://github.com/coveo/search-ui/blob/master/src/ui/Quickview/Quickview.ts